src/core/lib/security/transport/security_handshaker.cc

```diff
@@ -585,13 +586,14 @@ void SecurityHandshaker::DoHandshake(grpc_tcp_server_acceptor* /*acceptor*/,
 
 class FailHandshaker : public Handshaker {
  public:
+  explicit FailHandshaker(absl::Status status) : status_(std::move(status)) {}
   const char* name() const override { return "security_fail"; }
   void Shutdown(grpc_error_handle /*why*/) override {}
   void DoHandshake(grpc_tcp_server_acceptor* /*acceptor*/,
                    grpc_closure* on_handshake_done,
                    HandshakerArgs* args) override {
-    grpc_error_handle error =
-        GRPC_ERROR_CREATE("Failed to create security handshaker");
+    grpc_error_handle error = GRPC_ERROR_CREATE(absl::StrCat(
```

Comment:
Once you make the changes I suggested below, we won't need to do this here.  We can remove the local `error` variable and just directly use `status_` in its place.

Reply:
We don't need to call `GRPC_ERROR_CREATE()` at all.  Just pass `status_` to `ExecCtx::Run()` instead of `error`.

Note that `grpc_error_handle` used to be a custom type, but these days it's actually a typedef for `absl::Status`.  But there's a lot of legacy machinery for adding structured annotations to the `absl::Status` to simulate the legacy behavior of `grpc_error_handle`, and we're trying to slowly get rid of that tech debt in the code base.  `GRPC_ERROR_CREATE()` is one of the functions that add those legacy annotations, which is why I'm saying that we should get rid of it here.

Reply:
> these days it's actually a typedef for absl::Status

Ah this was my missing link, changed